### PR TITLE
Add insert() method to TimeInfo to allow Table row insertion

### LINKED ISSUE #7894: Exception when trying to add a row to a Table with a Time mix-in column
Adding a row to a Table with a Time mix-in column doesn't seem to work:

```python
In [44]: from astropy.time import Time

In [45]: from astropy.table import Table

In [46]: import numpy as np

In [47]: time = Time(np.arange(10), format='unix')

In [48]: t = Table({'time': time, 'flux': np.arange(10)})

In [49]: t
Out[49]: 
<Table length=10>
       time        flux
      object      int64
----------------- -----
              0.0     0
1.000000000001755     1
1.999999999998714     2
3.000000000000469     3
4.000000000002224     4
4.999999999999183     5
6.000000000000938     6
6.999999999997897     7
7.999999999999652     8
9.000000000001407     9

In [50]: t.add_row({'time': Time.now(), 'flux': 10})
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/Dropbox/Code/Astropy/astropy/astropy/table/table.py in insert_row(self, index, vals, mask)
   2343 
-> 2344                 newcol = col.insert(index, val, axis=0)
   2345                 if not isinstance(newcol, BaseColumn):

~/Dropbox/Code/Astropy/astropy/astropy/time/core.py in __getattr__(self, attr)
   1426             # Should raise AttributeError
-> 1427             return self.__getattribute__(attr)
   1428 

AttributeError: 'Time' object has no attribute 'insert'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-50-b77a2a3b5ec6> in <module>()
----> 1 t.add_row({'time': Time.now(), 'flux': 10})

~/Dropbox/Code/Astropy/astropy/astropy/table/table.py in add_row(self, vals, mask)
   2232              3   6   9
   2233         """
-> 2234         self.insert_row(len(self), vals, mask)
   2235 
   2236     def insert_row(self, index, vals=None, mask=None):

~/Dropbox/Code/Astropy/astropy/astropy/table/table.py in insert_row(self, index, vals, mask)
   2366         except Exception as err:
   2367             raise ValueError("Unable to insert row because of exception in column '{0}':\n{1}"
-> 2368                              .format(name, err))
   2369         else:
   2370             self._replace_cols(columns)

ValueError: Unable to insert row because of exception in column 'time':
'Time' object has no attribute 'insert'
```

### PR DESCRIPTION
**Pull request by taldcroft**
Closes #7894

### CODE REVIEW

**Comment by taldcroft** [comment]
```
In [6]: tm = Time(['2018:001', '2018:002'])

In [7]: tbl = Table([tm])

In [8]: tbl.insert_row(0, (Time('2010:123'),))

In [9]: tbl
Out[9]: 
<Table length=3>
         col0        
        object       
---------------------
2010:123:00:00:00.000
2018:001:00:00:00.000
2018:002:00:00:00.000
```

File: astropy/time/core.py [hunk 1]
```diff
@@ -817,6 +863,89 @@ def masked(self):
...
+        axis : int, optional
+            Axis along which to insert ``values``.  Default is 0, which is the
+            only allowed value and will insert a row.
+
+        Returns
+        -------
+        out : `~astropy.time.Time` subclass
+            New time object with inserted value(s)
+
+        Examples
```
Comment (taldcroft) [inline astropy/time/core.py]: Ooops, leftover, will remove.

**Comment by astrofrog** [comment]
@taldcroft - there are still a couple of sphinx warnings:

```
/home/circleci/project/docs/api/astropy.time.Time.rst:7: WARNING: py:obj reference target not found: np.insert
/home/circleci/project/docs/api/astropy.time.TimeDelta.rst:7: WARNING: py:obj reference target not found: np.insert
```

File: astropy/time/core.py [hunk 2]
```diff
@@ -817,6 +863,79 @@ def masked(self):
...
+
+        Returns
+        -------
+        out : `~astropy.time.Time` subclass
+            New time object with inserted value(s)
+
+        """
+        # Validate inputs: obj arg is integer, axis=0, self is not a scalar, and
+        # input index is in bounds.
+        if not isinstance(obj, (int, np.integer)):
```
Comment (mhvk) [inline astropy/time/core.py]: The secure, duck-typing way to get an index is
```
idx0 = operator.index(obj)
```
(can be within `try/except` if you want a different error message)

**Comment by astrofrog** [comment]
@taldcroft - can you rebase to get rid of the merge conflict?

**Comment by astrofrog** [comment]
I'll rebase so that we can include it before branching v3.1.x

**Pull request merged**